Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -58,76 +134,97 @@ namespace AzToolsFramework
+            // determine where to put this new row in the main DPE layout
+            auto newRow = new DPERowWidget(m_depth + 1);
+            DPERowWidget* priorWidgetInLayout = nullptr;
 
-            // if we found a valid handler, grab its widget and add it to our column layout
-            if (handlerId)
+            // search for an existing row sibling with a lower dom index
+            for (int priorWidgetIndex = domIndex - 1; priorWidgetInLayout == nullptr && priorWidgetIndex >= 0; --priorWidgetIndex)
             {
-                // store, then reference the unique_ptr that will manage the handler's lifetime
-                auto handler = dpeSystem->CreateHandlerInstance(handlerId);
-                handler->SetValueFromDom(childValue);
-                addedWidget = handler->GetWidget();
-                m_widgetToPropertyHandler[addedWidget] = AZStd::move(handler);
-                layoutForWidget = m_columnLayout;
+                priorWidgetInLayout = qobject_cast<DPERowWidget*>(m_domOrderedChildren[priorWidgetIndex]);
             }
-        }
-        else
-        {
-            AZ_Assert(0, "unknown node type for DPE");
-        }
 
-        // insert this new widget into the dom order, even if it's an empty widget to ensure the order is correct
-        auto insertIterator = m_domOrderedChildren.begin() + domIndex;
-        insertIterator = m_domOrderedChildren.insert(insertIterator, addedWidget);
+            // if we found a prior DPERowWidget, put this one after the last of its children,
+            // if not, put this new row immediately after its parent -- this
+            if (priorWidgetInLayout)
+            {
+                priorWidgetInLayout = priorWidgetInLayout->GetLastDescendentInLayout();
+            }
+            else
+            {
+                priorWidgetInLayout = this;
+            }
+            m_domOrderedChildren.insert(m_domOrderedChildren.begin() + domIndex, newRow);
+            GetDPE()->addAfterWidget(priorWidgetInLayout, newRow);
 
-        // properly place the new widget in its given layout, correctly ordered by DOM index
-        if (addedWidget && layoutForWidget)
+            // if it's a row, recursively populate the children from the DOM array in the passed value
+            newRow->SetValueFromDom(childValue);
+        }
+        else
         {
-            // search subsequent dom entries for the first occurrence of a widget in the same layout,
-            // then insert this new widget right before it. If there aren't any, then append this to the layout
-            ++insertIterator;
-            bool inserted = false;
-            while (!inserted && insertIterator != m_domOrderedChildren.end())
+            QWidget* addedWidget = nullptr;
+            if (childType == AZ::Dpe::GetNodeName<AZ::Dpe::Nodes::Label>())
+            {
+                auto labelString = childValue[AZ::DocumentPropertyEditor::Nodes::Label::Value.GetName()].GetString();
```

Comment:
This will crash if a label is missing text for whatever reason, I'd suggest using this to safely retrieve the value and default to an empty label:
`auto labelString = AZ::Dpe::Nodes::Label.ExtractFromDomNode(childValue).value_or("")`

Reply:
done, thanks!